crates/holochain/src/sweettest/sweet_consistency.rs

```diff
@@ -32,60 +32,174 @@ impl DurationOrSeconds {
+    // Declare op hash lists here so they can be accessed for reporting after timeout.
+    let mut rows_per_db = Vec::new();
+    let result = tokio::time::timeout(timeout, async {
+        loop {
+            tokio::time::sleep(Duration::from_millis(500)).await;
+            // Create query for each DHT DB.
+            let queries = cells.iter().map(|(_, dht_db)| {
+                dht_db.read_async(|txn| {
+                    let mut stmt = txn
+                        .prepare(
+                            "\
+                            SELECT DhtOp.hash, DhtOp.type, DhtOp.when_integrated, Action.seq, Action.author
+                            FROM DhtOp
+                            JOIN Action ON DhtOp.action_hash = Action.hash",
+                        )
+                        .unwrap();
+                    let mut rows = stmt.query([]).unwrap();
+                    let mut values = Vec::new();
+                    while let Some(row) = rows.next().unwrap() {
+                        let hash = row.get_unwrap::<_, DhtOpHash>(0);
+                        let op_type = row.get_unwrap::<_, DhtOpType>(1);
+                        let when_integrated = row.get_unwrap::<_, Option<Timestamp>>(2);
+                        let action_seq = row.get_unwrap::<_, u32>(3);
+                        let author = row.get_unwrap::<_, AgentPubKey>(4);
+                        values.push(DhtOpRow {
+                            hash,
+                            op_type,
+                            action_seq,
+                            author,
+                            when_integrated,
+                        });
+                    }
+                    Ok::<_, DatabaseError>(values)
+                })
+            });
+            // Execute queries in parallel.
+            rows_per_db = futures::future::join_all(queries)
+                .await
+                .into_iter()
+                .collect::<Result<_, _>>()
+                .unwrap();
+            // Build a set of all op hashes and create lists of hashes for each DHT DB.
+            let mut all_hashes = HashSet::new();
+            let mut hash_lists = Vec::new();
+            rows_per_db
+                .clone()
+                .into_iter()
+                .enumerate()
+                .for_each(|(index, dht_op_rows)| {
+                    tracing::debug!(
+                        "Agent {} with key {} has {} ops in their DHT DB",
+                        index,
+                        cells[index].0,
+                        dht_op_rows.len()
+                    );
+                    let mut hash_list = Vec::new();
+                    for row in dht_op_rows {
+                        // If any op is not yet integrated, continue to the next loop iteration.
+                        if row.when_integrated.is_none() {
+                            tracing::trace!("Unintegrated op found, sleeping...");
+                            continue;
```

Comment:
This `continue` applies to the most deeply-nested for loop, right? In that case it doesn't look like its sleeping so that can be removed from the trace log.

Or is this intended to apply to the top level `loop`, in which case I think we need to use a loop label to specify that.